packages/lit-html/src/lib/lit-html.ts

```diff
@@ -65,7 +63,14 @@ const isPrimitive = (value: unknown): value is Primitive =>
  *    * (") then any non-("), or
  *    * (') then any non-(')
  */
-const lastAttributeNameRegex = /([ \x09\x0a\x0c\x0d])([^\0-\x1F\x7F-\x9F "'>=/]+)([ \x09\x0a\x0c\x0d]*=[ \x09\x0a\x0c\x0d]*(?:[^ \x09\x0a\x0c\x0d"'`<>=]*|"[^"]*|'[^']*))$/;
+
+// These regex strings are used in the HTML scanner in Template. Keep them as
+// top-level variables. Terser will not inline them into a regex literal if
+// they are declared in the Template constructor.
+const SPACE_CHAR = `[ \\x09\\x0a\\x0c\\x0d]`;
+const ATTR_VALUE_CHAR = `[^ \\x09\\x0a\\x0c\\x0d"'\`<>=]`;
+const NAME_CHAR = `[^\\0-\\x1F\\x7F-\\x9F "'>=/]`;
+const TAG_END = `>|${SPACE_CHAR}(${NAME_CHAR}+)(${SPACE_CHAR}*=${SPACE_CHAR}*(?:${ATTR_VALUE_CHAR}|("|')|$))`;
```

Comment:
Note I also optimized the groups here to have fewer groups and one level less nesting. Hopefully it's faster.